merge/list_test.go

```diff
@@ -0,0 +1,282 @@
+package merge_test
+
+import (
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/typed"
+
+	"sigs.k8s.io/structured-merge-diff/merge"
+)
+
+func TestInlineListMerge(t *testing.T) {
+	parser, err := typed.NewParser(`types:
+- name: lists
+  struct:
+    fields:
+    - name: items
+      type:
+        list:
+          elementType:
+            scalar: string
+          elementRelationship: associative`)
+	if err != nil {
+		t.Fatalf("Failed to create parser: %v", err)
+	}
+
+	t.Run("disjointItemsOrder", func(t *testing.T) {
+		state := &State{
+			Updater:  &merge.Updater{},
+			Parser:   parser,
+			Typename: "lists",
+		}
+
+		lhs := typed.YAMLObject(`
```

Comment:
lhs and rhs make little sense in this package I think, since it's about apply multiple operations in a row.